packages/common/src/streams.ts

```diff
@@ -29,12 +32,21 @@ export const streamSize = async (stream: Readable): Promise<number> => {
   return size
 }
 
-export const streamToBytes = async (stream: Readable): Promise<Uint8Array> => {
-  const bufs: Buffer[] = []
-  for await (const bytes of stream) {
-    bufs.push(bytes)
+export const streamToBytes = async (
+  stream: Iterable<Uint8Array> | AsyncIterable<Uint8Array>,
+): Promise<Buffer> => {
+  const chunks: Uint8Array[] = []
+  let totalLength = 0 // keep track of total length for Buffer.concat
+  for await (const chunk of stream) {
+    if (chunk instanceof Uint8Array) {
+      chunks.push(chunk)
+      totalLength += Buffer.byteLength(chunk)
+    } else {
+      throw new TypeError('expected Uint8Array')
+    }
   }
-  return new Uint8Array(Buffer.concat(bufs))
+  // Note Buffer is a subclass of Uint8Array
+  return Buffer.concat(chunks, totalLength)
```

Comment:
any reason for changing the return type here?

i've been bit by subtle differences between Buffers & Uint8Arrays before & kinda prefer using just one or the other in most of our stuff is possible. or at least differentiating between the two and saying "bytes" for Uint8Arrays and "buffer" for Buffers

Reply:
`Buffer` is a sub-class of `Uint8Array`. So it kinda makes no sense to wrap a Buffer within an `Uint8Array` since we loose functionalities & we create two objects instead of one.
Since this function is only used in the pipethrough code, I will rename the fn.

Reply:
Note that `common-web` already has a `streamToBuffer` (that actually performs a `AsyncIterable<Uint8Array>` to `Uint8Array`). I think it's best if we don't change that function's name so I named this one to `streamToNodeBuffer` ...